File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        numbers = set()

        for i in nums:
            numbers.add(i)

        largest = 0
        for i in numbers:
            if i-1 not in numbers:
                currNum = i
                currCount = 1

                while currNum+1 in numbers:
                    currCount += 1
                    currNum += 1

                largest = max(currCount, largest)

        return largest


        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;

        unordered_set<int> st;
        int maxConsec = 0;

        for(int i=0;i<nums.size();i++)
        {
            st.insert(nums[i]);
        }

        for(int number: st)
        {
            if(!st.contains(number - 1))
            {
                int currNum = number;
                int currCount = 1;

                while(st.contains(currNum + 1))
                {
                    currNum = currNum + 1;
                    currCount += 1;
                }

                maxConsec = max(currCount, maxConsec);
            }

            
        }

        return maxConsec;
    }
};